client/routes/workflow/components/cytoscape-graph.vue

```diff
@@ -0,0 +1,237 @@
+      });
+    },
+    updateChildBtn(node) {
+      let nodeData = node.data();
+      if (nodeData.childRoute) {
+        store.commit("childRoute", {
+          route: nodeData.childRoute,
+          btnText: "To child"
+        });
+      } else if (nodeData.newExecutionRunId) {
+        store.commit("childRoute", {
+          route: nodeData.newExecutionRunId,
+          btnText: "Next execution"
+        });
+      } else {
+        store.commit("toggleChildBtn");
+      }
+    },
+    selectNode(id) {
+      //Deselect all previously selected nodes
+      cy.$(":selected").deselect();
+
+      //Mark current node as selected
+      let node = cy.elements("node#" + id);
+      node.select();
+
+      this.updateChildBtn(node);
+      this.zoomToNode(node);
+    },
+    async buildTree() {
+      this.events.forEach(event => {
+        let {
+          parentWorkflowExecution,
+          previousExecutionRunId,
+          newExecutionRunId,
+          status,
+          childRoute
+        } = getEventConnections(event, this.events);
+
+        //We are viewing a child workflow, show parent btn
+        if (previousExecutionRunId) {
+          store.commit("previousExecutionRoute", previousExecutionRunId);
+        } else if (parentWorkflowExecution) {
+          store.commit("parentRoute", parentWorkflowExecution);
+        }
+
+        this.nodes.push({
+          data: {
+            id: event.eventId,
+            name: event.eventType,
+            childRoute: childRoute,
+            newExecutionRunId: newExecutionRunId,
+            status: status
+          }
+        });
+      });
+      //Set the direct and inferred relationships
+      this.events.forEach(node => {
+        this.setDirectAndInferred(node);
+      });
+
+      //Set the chronological relationships.
+      //If the node is not referred to as a parent it should be connected back to the graph with a chron child
+      this.events.forEach(node => {
+        if (!this.parentArray.includes(node.eventId)) {
+          this.setChron(node);
+        }
+      });
+    },
+    setDirectAndInferred(node) {
+      let nodeId = node.eventId,
+        { parent, inferredChild } = getEventConnections(node, this.events);
+      if (parent) {
+        this.parentArray.push(parent);
+        this.edges.push({
+          data: { source: parent, target: nodeId, type: "direct" }
+        });
+      }
+      if (inferredChild) {
+        this.parentArray.push(nodeId);
+        this.edges.push({
+          data: { source: nodeId, target: inferredChild, type: "inferred" }
+        });
+      }
+    },
+    setChron(node) {
+      let nodeId = node.eventId,
+        { chronologicalChild } = getEventConnections(node, this.events);
+      if (chronologicalChild) {
+        this.edges.push({
+          data: {
+            source: nodeId,
+            target: chronologicalChild,
+            type: "chronological"
+          }
+        });
+      }
+    },
+    async viewInit() {
```

Comment:
is this function actually asynchronous?

Reply:
Nope, good catch. That was left from an old functionality. I removed it!